proxy.go

```diff
@@ -73,7 +73,7 @@ func Join(local, remote net.Conn, log logging.Logger) {
 
 	transfer := func(side string, dst, src net.Conn) {
 		log.Debug("proxing %s -> %s", src.RemoteAddr(), dst.RemoteAddr())
-
+		defer dst.Close()
```

Comment:
> Should I still change this line to CloseRead anyway?

I responded without actually reading the code - just from top of my head. We have two goroutines - one that reads from src and second one that writes to it. If you Close() the src as soon as you stopped reading, the other goroutine might be still writing to it. This is why we need the distinction between Close and CloseRead and CloseWrite.

I really haven't got around fixing and testing this, but will do eventually if no-one does it earlier. 😉